### ISSUE TITLE
Issues about surface representation

### ISSUE THREAD

**Issue by lhanappa** [SystemErrorWang/White-box-Cartoonization#12]
Hi, 
An interesting topic and a great model! Thanks for sharing.
Here is an issue about surface representation after I read the paper: How do you define the surface representation _F_dgf_? I didn't find the definition in section3.1 and after. 

I'm not quite expert with tf v1, if my understand corrects,  _L_total_ in the section 3.4 paper is related to "g_loss_total = 1e4*tv_loss + 1e-1*g_loss_blur + g_loss_gray + 2e2*recon_loss" line 77, train.py, Right? Well, how to map the four items to the section 3.4?

One more general question,  I think this model contributes a combination of losses functions. So my question is why you design these 5 items or 3 parts: structure, texture, and surface? Is there any reference supporting, because I always confused with these terminologies.

Thanks.

**Comment by SystemErrorWang** [comment]
@lhanappa F_dgf means differentiable guided filter，as the name shows, it extracts surface representation with a differentiable guided filter. 
The loss function consists of 5 terms, g_loss_blur means surface representation, g_loss_gray means texture representation, superpixel_loss means structure representation, the rest of loss terms are described in the main paper. It's true that the name of loss terms differ from the terminologies in the paper, I think I'd better modify them.

**Comment by lhanappa** [comment]
@SystemErrorWang thanks. Can't wait for.

Could you please give some details about the differentiable guided filter, such as the expressions or the link of reference?

**Comment by SystemErrorWang** [comment]
@lhanappa for code, it is contained in this repo. for githun repo: https://github.com/wuhuikai/DeepGuidedFilter

**Comment by lhanappa** [comment]
thanks

**Comment by Xinxiang7** [comment]
Hi, @SystemErrorWang I tried the guided filter individually. 
If it takes an image I as input and itself as guide map, returns the same picture instead of extracted surface representation, just as follows:
![gf3](https://user-images.githubusercontent.com/24989537/85005021-d24baf00-b18a-11ea-9c10-47c4a548115c.png)
I also tried code of github repo:https://github.com/wuhuikai/DeepGuidedFilter. It's the same effect.
I read the paper `Fast End-to-End Trainable Guided Filter` and find that the effect of output depends guided map. If the guided map is high resolution, the output will be high resolution, vice versa. So how you get the surface reprentationn with  textures and details removed. Looking forward to your reply!
